experimental/uefi/app/src/remote_attestation.rs

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Server-side implementation of the bidirectional gRPC remote attestation handshake
+//! protocol.
+//!
+//! A simplified version of the implementation from the `grpc_unary_attestation`
+//! crate. TODO(#2741): Refactor this to share more code between the two runtimes.
+
+use alloc::vec::Vec;
+use oak_remote_attestation_sessions::{SessionId, SessionState, SessionTracker};
+
+/// Number of sessions that will be kept in memory.
+const SESSIONS_CACHE_SIZE: usize = 10000;
+
+pub struct AttestationHandler<F> {
+    session_tracker: SessionTracker,
+    request_handler: F,
+}
+
+const MOCK_TEE_CERTIFICATE: [u8; 0] = [];
+const MOCK_ADDITIONAL_INFO: [u8; 0] = [];
+
+impl<F> AttestationHandler<F>
```

Comment:
This feels like this is more generic functionality, and could be reused for different scenarios. It might be a good idea to move it somewhere more reusable (either `remote_attestation_sessions` (my preference) or the new runtime crate from #2745, or even yet another crate (although that is probably overkill)).

Reply:
As you mentioned in https://github.com/project-oak/oak/pull/2742#discussion_r856083684 this can be done later as part of #2471